src/main/java/com/marketdata/MarketSimulatorBackend/service/MockStockProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.marketdata.MarketSimulatorBackend.service;
+
+import com.marketdata.MarketSimulatorBackend.model.StockPrice;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+@Profile("mock")
+public class MockStockProvider implements StockServiceProvider{
```

Comment:
good job, very nice way of using interface.

In clean architecture, the interface itself will be in the service layer, and the implementations will be in the infrastructure layer.